src/Agent.Worker/JobExtension.cs

```diff
@@ -564,6 +556,18 @@ public async Task<List<IStep>> InitializeJob(IExecutionContext jobContext, Pipel
             }
         }
 
+        private void PublishAgentShutdownTelemetry(IExecutionContext jobContext, IExecutionContext childContext)
+        {
+            var telemetryData = new Dictionary<string, string>
+            {
+                { "JobId", childContext?.Variables?.System_JobId?.ToString() ?? string.Empty },
+                { "JobResult", TaskResult.Failed.ToString() },
+                { "TracePoint", "110" },
```

Comment:
[Not blocking] It would be good to place all such traces together in some class for better visibility